src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+
+        # We want to run a python interpreter somewhere...
+        cmd: Sequence[str] = ('python3', '-ic', '# cockpit-bridge')
+        env: Sequence[str] = ()
+
+        # Remote host?  Wrap command with SSH
+        if self.destination != 'localhost':
+            ssh_askpass = ensure_ferny_askpass()
+            env = (
+                f'SSH_ASKPASS={ssh_askpass!s}',
+                'DISPLAY=x',
+                'SSH_ASKPASS_REQUIRE=force',
+            )
+            cmd = ('ssh', self.destination, shlex.join(cmd))
+
+        # Running in flatpak?  Wrap command with flatpak-spawn --host
+        if os.path.exists('/.flatpak-info'):
+            cmd = ('flatpak-spawn', '--host',
+                   *(f'--env={kv}' for kv in env),
+                   *cmd)
+            env = ()
+
+        logger.debug("Launching command: cmd=%s env=%s", cmd, env)
+        transport = await self.spawn(cmd, env, stderr=agent, start_new_session=True)
+
+        # Send the first-stage bootloader
+        stage1 = bootloader.make_bootloader([
+            ('report_exists', [list(get_interesting_files())]),
+            ('boot_xz', [beipack_filename, len(bridge_beipack_xz)])
+        ], gadgets=BEIBOOT_GADGETS)
+        transport.write(stage1.encode())
+
+        # Wait for "init" or error, handling auth and beiboot requests
+        await agent.communicate()
+
+        return transport
+
+
+class SshBridge(Router):
+    packages: Packages
+    ssh_peer: SshPeer
+
+    def __init__(self, args: argparse.Namespace):
+        self.ssh_peer = SshPeer(self, args.destination)
+
+        super().__init__([
+            DefaultRoutingRule(self, self.ssh_peer),
+        ])
+
+    def do_send_init(self):
+        pass  # wait for the peer to do it first
+
+    def do_init(self, message):
+        self.ssh_peer.write_control(**message)
+
+
+async def run(args) -> None:
+    logger.debug("Hi. How are you today?")
+
+    bridge = SshBridge(args)
+    StdioTransport(asyncio.get_running_loop(), bridge)
+
+    try:
+        message = await bridge.ssh_peer.start()
+        bridge.write_control(**dict(message, packages={p: None for p in bridge.packages.packages}))
+        bridge.ssh_peer.thaw_endpoint()
+    except ferny.InteractionError as exc:
+        sys.exit(str(exc))
+    except CockpitProblem as exc:
+        bridge.write_control(command='init', problem=exc.problem, **exc.kwargs)
+        return
+
+    logger.debug('Startup done.  Looping until connection closes.')
+    try:
+        await bridge.communicate()
+    except BrokenPipeError:
+        # not unexpected if the peer doesn't hang up cleanly
+        pass
+
+
+def main() -> None:
+    polyfills.install()
+
+    parser = argparse.ArgumentParser(description='cockpit-bridge is run automatically inside of a Cockpit session.')
+    parser.add_argument('--debug', action='store_true')
+    parser.add_argument('destination', help="Name of the remote host to connect to")
```

Comment:
perhaps mention `localhost`? `127.0.0.1` or empty string are other plausible values (but they don't work)

Reply:
This is not really meant for human consumption, but then why have help at all, right?  Will add it.

Reply:
Done